lib/cartopy/tests/test_linear_ring.py

```diff
@@ -23,25 +23,21 @@ def test_cuts(self):
         assert len(multi_line_string.geoms) > 1
         assert not rings
 
-        def assert_intersection_with_boundary(segment_coords):
-            # Double the length of the segment.
-            start = segment_coords[0]
-            end = segment_coords[1]
-            end = [end[i] + 2 * (end[i] - start[i]) for i in (0, 1)]
-            extended_segment = sgeom.LineString([start, end])
-            # And see if it crosses the boundary.
-            intersection = extended_segment.intersection(projection.boundary)
-            assert not intersection.is_empty, 'Bad topology near boundary'
-
-        # Each line resulting from the split should start and end with a
-        # segment that crosses the boundary when extended to double length.
+        def assert_close_to_boundary(xy):
```

Comment:
The latest commit changes this. As far as I can tell we don't extend a line segment by 2x the distance to attach to a boundary anywhere, rather we just project along the boundary to find the closest point. So really we just want to be "close" and not be in the middle of the domain. This is relevant because some segments may be very close to the boundary, but when extended they don't go very far because the points are too closely spaced.